#Scenes/SceneScripts/SplashScene.gd

```diff
@@ -0,0 +1,25 @@
+extends Control
+
+@export var splash_image: TextureRect
+@export var uuuu_sfx: AudioStreamPlayer2D
+
+var tween: Tween
+
+func _ready() -> void:
+	tween = create_tween()
+	var original_scale: Vector2 = splash_image.scale
+	# Animation to slingshot the logo, scale low the image with a spring like animation 
+	# then scale it back to original size with an elastic like animation
+	# Play Fauna uuuu-ing in-between, then fade splash away and navigate to main menu when done
+	tween.tween_property(splash_image, "scale", Vector2(0.30, 0.30), 1.5).set_trans(Tween.TRANS_SPRING).set_ease(Tween.EASE_IN)
+	tween.tween_callback(_play_sfx)
```

Comment:
not sure there is a need to repeatedly call the SFX (since if I understood correctly, that's what the callback does)

Reply:
Both callbacks shouldn't be repeatedly calling? Where are you reading that at?

Reply:
Ah nvm I misread [the doc](https://docs.godotengine.org/en/stable/classes/class_tween.html#class-tween-method-tween-callback)
![image](https://github.com/Saplings-Projects/1M_sub/assets/49365680/f0c668c3-24bb-4f2f-97c2-149e70f820f2)
didn't read the example exactly and I assumed the callback was to do a repeated call but that's not the case. What is callback used for, sequential execution ?

Reply:
Callback is to fire some function to call when we hit that section of the tween. In this case we call the _play_sfx() function after we do the initial scale. Similar to how signals work and firing a signal's callback when it's been emitted.